### ISSUE TITLE
Wrong getSkinUrl() value in ESI context when skin files location differs from layout files location

### ISSUE THREAD

**Issue by ceckoslab** [nexcess/magento-turpentine#971]
In rare cases we may want to use same structure (layout and templates) but different colors and images aka. skin per store view. Magento platform has this feature out of the box and this can be easily configured in Magento backoffice.

![turpentine-skin-path-bug](https://cloud.githubusercontent.com/assets/1024001/10268459/7cc73210-6ab9-11e5-8924-9a61049c0541.png)

Unfortunately when we use Turpentine extension with ESI caching enabled the scenario described above  doesn't work.

If we accept that we have the configuration show in the screenshot above then when we use $this->getSkinUrl() inside a block that is about to be created in ESI context we always will get file "default" rather than from "red_skin".

I believe that this bug was not discovered till now because the store view configuration that I shown is not so common.

I am preparing PR that that could solve this problem.

**Comment by miguelbalparda** [comment]
Have you seen https://github.com/nexcess/magento-turpentine/issues/564 ?

**Comment by ceckoslab** [comment]
@miguelbalparda I read what @robin31 purposes in #564 but I explain in #972 that we don't need to set anything to Mage::getSingleton( 'core/design_package' ) because when Magento knows the store id then everything needed is read from configuration.

Magento does everything automatically so any of those lines is not needed:

```
Mage::getSingleton( 'core/design_package' )
->setPackageName( $esiData->getDesignPackage() )
->setTheme( $esiData->getDesignTheme() )
->setTheme('layout', Mage::getStoreConfig('design/theme/layout'))
->setTheme('template', Mage::getStoreConfig('design/theme/template'))
->setTheme('skin', Mage::getStoreConfig('design/theme/skin'))
->setTheme('locale', Mage::getStoreConfig('design/theme/locale'));
```

Of course my solution needs code review and second opinion because it may break something in another context.

**Comment by miguelbalparda** [comment]
Fixed by #972
